crossbar/router/broker.py

```diff
@@ -171,13 +171,17 @@ def _filter_publish_receivers(self, receivers, publish):
 
         # if "eligible_authid" we only accept receivers that have the correct authid
         if publish.eligible_authid:
+            eligible = set()
             for aid in publish.eligible_authid:
-                receivers = receivers & self._router._authid_to_sessions.get(aid, set())
```

Comment:
I guess, above code at least has a better chance of being optimized, as `set.union` explicitly can work on a list of set-valued arguments. As said, I don't know if say PyPy will be able to optimize away temporary objects.

The time complexity of computing the union set of a list/set of sets should be O(len(s1) + len(s2) + ... len(sn)) ..